From 775b9a35f64c16e45cdf1d2001faae323d67afbd Mon Sep 17 00:00:00 2001
From: Jiani Liu <quic_jianil@quicinc.com>
Date: Fri, 7 Apr 2023 11:22:42 +0800
Subject: [PATCH] ath10k: fix fwlog carsh

The skb is released twice when WMI_TLV_DEBUG_MESG_EVENTID message
received.

Data bus error, epc == 832117c4, ra == 832117c4
Oops[#1]:
CPU: 0 PID: 620 Comm: kmodloader Not tainted 3.18.29 #4
task: 83010ed0 ti: 83006000 task.ti: 83006000
$ 0 : 00000000 804a0000 deadc0de 00000000
$ 4 : b0034410 83210000 00000000 00034414
$ 8 : 00000000 00000004 00000000 30302e30
$12 : 00000000 83817724 00000000 20303030
$16 : 831493c0 00034410 ffff0000 00000000
$20 : 00034410 838b0468 83214ae8 00000017
$24 : 00000003 8321173c
$28 : 83006000 83007bf0 00000000 832117c4
Hi : 00000009
Lo : 00000006
epc : 832117c4 ath10k_pci_hif_exchange_bmi_msg+0x7bc/0xd60 [ath10k_pci]
Not tainted
ra : 832117c4 ath10k_pci_hif_exchange_bmi_msg+0x7bc/0xd60 [ath10k_pci]
Status: 1000fc03 KERNEL EXL IE
Cause : 1080001c
PrId : 00019374 (MIPS 24Kc)

Change-Id: I1a195fdb1116418fd356fe0e4fdd85031b1301f5
Signed-off-by: Miaoqing Pan <miaoqing@codeaurora.org>
Signed-off-by: Jiani Liu <quic_jianil@quicinc.com>

diff --git a/drivers/net/wireless/ath/ath10k/wmi-tlv.c b/drivers/net/wireless/ath/ath10k/wmi-tlv.c
index d97b33f..515a379 100644
--- a/drivers/net/wireless/ath/ath10k/wmi-tlv.c
+++ b/drivers/net/wireless/ath/ath10k/wmi-tlv.c
@@ -663,7 +663,7 @@ static void ath10k_wmi_tlv_op_rx(struct ath10k *ar, struct sk_buff *skb)
 		break;
 	case WMI_TLV_DEBUG_MESG_EVENTID:
 		ath10k_wmi_event_debug_mesg(ar, skb);
-		break;
+		return;
 	case WMI_TLV_UPDATE_STATS_EVENTID:
 		ath10k_wmi_event_update_stats(ar, skb);
 		break;
-- 
2.7.4

